File: solutions/2026-01-05/solution.java
// Maximum Subarray
// Difficulty: Medium
// URL: https://leetcode.com/problems/maximum-subarray/

public int maxSubArray(int[] nums) {
    if (nums.length == 0) return 0;
    int maxSum = Integer.MIN_VALUE;
    int currentSum = 0;
    for (int num : nums) {
        currentSum = Math.max(num, currentSum + num);
        maxSum = Math.max(maxSum, currentSum);
    }
    return maxSum;
}

File: solutions/2026-01-05/solution.py
# Maximum Subarray
# Difficulty: Medium
# URL: https://leetcode.com/problems/maximum-subarray/

def maxSubArray(nums):
    if not nums:
        return 0
    max_sum = float('-inf')
    current_sum = 0
    for num in nums:
        current_sum = max(num, current_sum + num)
        max_sum = max(max_sum, current_sum)
    return max_sum